CantoboardFramework/Keyboard/InputController.swift

```diff
@@ -493,13 +493,21 @@ class InputController: NSObject {
         if Settings.cached.isMixedModeEnabled && state.inputMode == .chinese { state.inputMode = .mixed }
         if !Settings.cached.isMixedModeEnabled && state.inputMode == .mixed { state.inputMode = .chinese }
         
-        isImmediateMode =  Settings.cached.compositionMode == .immediate
+        isImmediateMode = state.inputMode == .english || Settings.cached.compositionMode == .immediate
         if isImmediateMode {
             if !(compositionRenderer is ImmediateModeCompositionRenderer) {
+                if let compositionRenderer = compositionRenderer, compositionRenderer.hasText {
```

Comment:
Could we move it to CompositionRenderer.deinit?

Reply:
Modified.